ui/admin/app/components/form/target/details/index.hbs

```diff
@@ -75,86 +81,95 @@
+    <F.HelperText>{{t 'form.session_max_seconds.help'}}</F.HelperText>
     {{#if @model.errors.session_max_seconds}}
-      <field.errors as |errors|>
+      <F.Error as |E|>
         {{#each @model.errors.session_max_seconds as |error|}}
-          <errors.message>{{error.message}}</errors.message>
+          <E.Message>{{error.message}}</E.Message>
         {{/each}}
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
-  <form.input
-    @name='session_connection_limit'
-    @type='number'
+  <Hds::Form::TextInput::Field
+    @isRequired={{false}}
+    @isOptional={{true}}
     @value={{@model.session_connection_limit}}
-    @label={{t 'form.session_connection_limit.label'}}
-    @error={{@model.errors.session_connection_limit}}
-    @helperText={{t 'form.session_connection_limit.help'}}
-    min='-1'
-    step='1'
-    as |field|
+    @isInvalid={{@model.errors.session_connection_limit}}
+    @type='text'
+    name='session_connection_limit'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'session_connection_limit')}}
+    as |F|
   >
+    <F.Label>{{t 'form.session_connection_limit.label'}}</F.Label>
+    <F.HelperText>{{t 'form.session_connection_limit.help'}}</F.HelperText>
     {{#if @model.errors.session_connection_limit}}
-      <field.errors as |errors|>
+      <F.Error as |E|>
         {{#each @model.errors.session_connection_limit as |error|}}
-          <errors.message>{{error.message}}</errors.message>
+          <E.Message>{{error.message}}</E.Message>
         {{/each}}
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
-  <form.input
-    @name='worker_filter'
-    @type='text'
+  <Hds::Form::TextInput::Field
+    @isRequired={{false}}
+    @isOptional={{true}}
     @value={{@model.worker_filter}}
-    @label={{t 'form.worker_filter.label'}}
-    @error={{@model.errors.worker_filter}}
-    @helperText={{t 'form.worker_filter.help'}}
-    @linkText={{t 'actions.learn-more'}}
-    @link={{doc-url 'target.worker-filters'}}
-    as |field|
+    @isInvalid={{@model.errors.worker_filter}}
+    @type='text'
+    name='worker_filter'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'worker_filter')}}
+    as |F|
   >
+    <F.Label>{{t 'form.worker_filter.label'}}</F.Label>
+    <F.HelperText>{{t 'form.worker_filter.help'}}</F.HelperText>
     {{#if @model.errors.worker_filter}}
-      <field.errors as |errors|>
+      <F.Error as |E|>
         {{#each @model.errors.worker_filter as |error|}}
-          <errors.message>{{error.message}}</errors.message>
+          <E.Message>{{error.message}}</E.Message>
         {{/each}}
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
-  <form.input
-    @name='default_port'
-    @type='number'
+  <Hds::Form::TextInput::Field
+    @isRequired={{true}}
```

Comment:
If default port is only required for `tcp` can we check what type of target we have to make it required only for `tcp`?

Reply:
Good idea. I can get that pushed up shortly.